src/code-considerations/using-unstable-lang/specialization.md

```diff
@@ -58,6 +58,39 @@ impl<T: Copy> RcFromSlice<T> for Rc<[T]> {
 
 Only specialization using the `min_specialization` feature should be used. The full `specialization` feature is known to be unsound.
 
+## Specialization attributes
+
+There are two unstable attributes that can be used to allow a trait bound in a specializing implementation that does not appear in the default implementation.
+
+`rustc_specialization_trait` restricts the implementations of a trait to be "always applicable". Implementing traits annotated with `rustc_specialization_trait` is unstable, so this should not be used on any stable traits exported from the standard library. `Sized` is an exception, and can have this attribute because it already cannot be implemented by an `impl` block.
+
+`rustc_unsafe_specialization_marker` allows specializing on a trait with no associated items. The attribute is `unsafe` because lifetime constraints from the implementations of the trait are not considered when specializing. In the following example, the specialized implementation is used for *all* shared reference types, not just those with `'static` lifetime.
```

Comment:
Am I understanding correctly that the point of this example is to show incorrect usage of `rustc_unsafe_specialization_marker`? If so we should probably note that more clearly.
```suggestion
`rustc_unsafe_specialization_marker` allows specializing on a trait with no associated items. The attribute is `unsafe` because lifetime constraints from the implementations of the trait are not considered when specializing. The following example demonstrates incorrect usage of `rustc_unsafe_specialization_marker`, the specialized implementation is used for *all* shared reference types, not just those with `'static` lifetime.
```

Reply:
I've tried to make it clearer that this example is showing a limitation of the specialization when using the attribute.